apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.ts

```diff
@@ -28,7 +34,29 @@ router.get(
       CourseRequestRowSchema,
     );
 
-    res.send(RequestCourse({ rows, resLocals: res.locals }));
+    let lti13Info: Lti13CourseRequestInput = null;
+    if (isEnterprise() && 'lti13_claims' in req.session) {
```

Comment:
When does this get removed from the session? My concern is that this would get stuck in the user's session and they'd see this modal forever, even if on future requests they aren't coming from the LTI course setup flow. Consider adding a query param to indicate that we're in that flow so that even if this happens to be present in the session, we otherwise won't show it.

Reply:
In a normal LTI 1.3 flow, we remove it from the session in the "course navigation" lookup before directing to the course instance. So if a course is already setup, it shouldn't be present if they go to the form after the normal flow.

If they don't have any course instances and they get the LTI 1.3 course navigation page that prompts for more info, the session isn't removed so it works when they click through to the request page.

Is that sufficient without adding query param flows too?

Reply:
I'm thinking of the case where someone lands in PL via LTI but doesn't finis the onboarding flow (e.g. because they have to request a course). If they then land on the course request page again via some independent path, we'd still show them the "you're coming via LTI!" which would be confusing. I think I'd still rather us be explicit about when we're requesting a course as part of the LTI flow. Does that concern make sense?

Reply:
The "con" to that approach is that if that new instructor in the onboarding flow clicks around to other places (like the homepage, or other courses, etc.) and gets to request course in a different route they won't get the autofill. I don't know what people would do in practice.

The LTI 1.3 session has a 1 hour timeout, and after that point it doesn't show. If the information is incorrect, it will let them edit it. I think this is useful enough (even if it was only filling in name/institution) that I wouldn't constrain it to just an LTI onboarding entry path.